packages/webcert/src/store/fmb/fmbMiddleware.ts

```diff
@@ -87,6 +143,20 @@ export const handleUpdateCertificateDataElement: Middleware<Dispatch> = ({ dispa
     }
 
     getFMBDiagnosisCodes(action.payload.value, getState().ui.uiFMB.fmbDiagnosisCodeInfo, dispatch)
+
+    if (
+      action.payload.value &&
+      getState().ui.uiCertificate.certificate &&
+      (action.payload.value.type === CertificateDataValueType.DATE_RANGE_LIST ||
+        action.payload.value.type === CertificateDataValueType.DIAGNOSIS_LIST)
+    ) {
+      getValidationForSickLeavePeriod(
+        action.payload.value,
+        getState().ui.uiCertificate.certificate.metadata.patient.personId.id,
```

Comment:
We should really avoid accessing other slices of states. Maybe we can consider keeping the information in the FMB-slice that we need instead of accessing the Certificate-slice. For example, we can get the patients personId when updateCertificate-action is dispatched.